### DISCUSSION TITLE
Savvycan and windows 10 connections,

### DISCUSSION THREAD

**Discussion by sdarbey** [collin80/SavvyCAN#559] [Q&A]
Hi there everybody and happy new year to you.

As a newbie I hope this is something very simple and as such I am reaching out for the tick box that makes Savvycan and GVRET work.
Ordered Candue unit from EVTV a few weeks ago and it arrived promptly. Saved the package for Christmas and started configuring just before the break.
Download all the drivers etc, from Collin's Github and the Due eventually uploaded and verified in Arduino IDE. All good so far.
Installed Savvycan on Three PC's 1 Home desktop and two laptops all running W10 Kernal 22H2.

Problem start when the Connection window is opened and loaded GVRET driver and set serial baud rate and can speed.
Console window opens and enabled, I cam also send hex codes and text ( Flicking led on Due works )
Driver says Connected. But no Can mesages appear when I select 'Start Capturing' have connected the two wires to Can0L and Can0H. ( works with my previous Can bus sniffer ) I set the can speed to 500000 and The baud rate to 115200. If I revert to my previous Can hacker unit works as soon as I attach the two wires.

I believe my new toy to be vastly superior.

All help hugely appreciated no matter how small.

Regards
stephen

**Comment by collin80** [comment]
What do you mean you set the baud rate to 115200? Savvycan doesnt ask you to set a serial speed as it's handled behind the scenes and always the same value. But, it makes sense if you're trying to also connect with a serial terminal to test things.

Are you using the original candue or an esp32due? The older CanDue actually is USB native and so no baud rate you set will make any difference. It always runs as fast as it can. The ESP32 based boards aren't USB native and run at a serial speed of 1000000.

**Reply by sdarbey** [reply]
Ah Thanks for getting back to me Colin.

Your are right. There is no baud rate setting, I confused this with my old unit. I open Savvycan and then go to New connection. I choose Serial GVRET and it correctly chooses Com3. The connection is added and after a few seconds it says Connected. Note it will only connect if I untick the 'Verify Validation of GVRET Connection' in the preferences section. I am using V208 Dec8 2022 of Savvycan. Below the connections window It says Speed and I say 500000 as I am presuming this is the canbus speed.  I have installed a few different versions on  two of my laptops with the same result. My can wires are connected to Can0L and Can0H on the Candue
My CanDue is connected to my PC via USB cable.

Any hints at all will be hugely appreciated no matter how silly I look.
Regards
Stephen
![Candue](https://user-images.githubusercontent.com/38258221/211931747-facf165c-959f-413d-9249-84c7d9ee13c3.jpg)

**Comment by collin80** [comment]
Ok, that's an older design but there's nothing wrong with it. That design is actually USB native and can sustain incredibly high bus load without dropping any frames. Anyway, the first thing I would do is re-flash the board to see if perhaps it isn't running the proper software or something got corrupted.  That you had to uncheck to verify the connection is telling. Without validation it'll say it is connected even if it really isn't working. There is a binary updater here: https://www.savvycan.com/GVRETupdate.zip unzip that, plug in the board, and run the appropriate script file for your OS. It appears you mentioned Windows so that'd be the .bat file for you. This should find the board and program it. Afterward you should disconnect USB, plug it back in, and try again. You should always have validation turned on. If it doesn't work with validation turned on then it probably really isn't working.

**Reply by sdarbey** [reply]
Thanks again Colin

I will upgrade this evening and try again over the weekend

Regards
Stephen

**Reply by sdarbey** [reply]
Hi again.

Here is what I get when  I run batch file.
I can confirm that Com3 is definitely working

C:\T@mp\GVRETupdate>.\GVRETupdate.bat

C:\T@mp\GVRETupdate>for /F "tokens=1-2 delims=- " %i in ('listComPorts.exe') do (
if "%j" == "Arduino" mode %i: baud=1200
 if "%j" == "Microsoft" mode %i: baud=1200
)

C:\T@mp\GVRETupdate>(
if "Arduino" == "Arduino" mode COM3: baud=1200
 if "Arduino" == "Microsoft" mode COM3: baud=1200
)

Status for device COM3:
-----------------------
    Baud:            1200
    Parity:          Even
    Data Bits:       7
    Stop Bits:       1
    Timeout:         OFF
    XON/XOFF:        OFF
    CTS handshaking: OFF
    DSR handshaking: OFF
    DSR sensitivity: OFF
    DTR circuit:     OFF
    RTS circuit:     ON


C:\T@mp\GVRETupdate>pause
Press any key to continue . . .

C:\T@mp\GVRETupdate>bossac -e -w -v -b GVRET.ino.bin
Auto scan for device failed
Try specifying a serial port with the '-p' option

C:\T@mp\GVRETupdate>pause
Press any key to continue . . .

C:\T@mp\GVRETupdate>

Regards
Stephen

**Comment by collin80** [comment]
Hmmm... maybe something got really jammed. Usually that doesn't happen but it can. One thing you can try is to remove the top board (carefully... the pins are kind of long and like to bend) and then hold down the erase button on the bottom board for 2 seconds (while the board is connected to USB). Then disconnect and reconnect. You should be able to flash the program while the top board is still off. This will totally zap whatever is currently flashed to the board. Sometimes that make life easier. Though, it just dawned on me another issue you may be having. The original Arduino Due which should be what the bottom board is, has two USB ports on it. We always use the "native" port. There is also a "programming" port. For some silly reason they don't seem to label them but I'm pretty sure the upper one is the native port (it's been a while since I used one of these). So, you might try that one instead. It still ought to have be able to use the bottom port to flash, hence calling it the "programming" port but it can't hurt to try the other one just to be sure.

**Reply by sdarbey** [reply]
Whoa. Bingo

Did exactly this and to no avail, then I tried 2nd port which registered as Com5 and managed to run GVretupdate from the command prompt and it successfully uploaded new firmware. Verified and compiled again and device added successfully. I then attached to a canbus to see lots of frames appearing. You're right about removing the EVTV board. It took me ages to get it mounted in the first place. Must have taken a knock in transit because some of the pins were askew.

Now all I have to do is figure out why com3 won't work, but I don't necessarily need it anyway.

Thank you so much. If you're ever in Ireland I'lll treat you to a few pints as a token of my appreciation. Now to get interrogating and learning.

Regards and many thanks.
Stephen
